### ISSUE TITLE
How to use Navigation to show a master-detail layout?

### ISSUE THREAD

**Issue by Zhuinden** [googlesamples/android-architecture-components#377]
It appears to me that you need to write your own navigator implementation that must handle the global `navigate()`  method.

Is it a recommendation of sorts that the current destination's ID is persisted in an internal fragment directly into the FragmentManager?

**Comment by KingIdee** [comment]
Hey @Zhuinden have you tried something? I'll like to see. I'm in need of this

**Comment by Zhuinden** [comment]
@KingIdee I have an idea, but I expect that because https://github.com/googlesamples/android-sunflower requires it, the Navigation team will figure something out for out-of-the-box managemnet

**Comment by KingIdee** [comment]
Okay. I hope we don't have to wait till io19 😄

**Comment by Zhuinden** [comment]
I saw https://proandroiddev.com/master-detail-views-with-navigation-components-a20405f31974 and it makes sense.

**Comment by PatrinAV** [comment]
Yeah. But what we can do if we need master/detail behavior not for tablet/phone but for portrait/landscape

**Comment by MarcusWolschon** [comment]
> Yeah. But what we can do if we need master/detail behavior not for tablet/phone but for portrait/landscape

Then use layout-w600dp (actual screen width in current orientation) instead of layout-sw600dp (shortest width).

**Comment by tiagomcouto** [comment]
hey there!! I have one question, how do I pass data from the Master to the Detail fragment. I noticed that doesn't work with safeArgs because we need to use 2 graphs so there is no action to be able to use safeArgs. Anyone knows how to do it?

**Comment by Zhuinden** [comment]
I don't think you need 2 graphs for a master detail, that sounds tricky to manage 🤔

**Comment by tiagomcouto** [comment]
how would you manage a master detail with only 1 graph :o ? using navigation. If you have a way let me know :P

**Comment by tiagomcouto** [comment]
because if I'm not mistaken you can't do master detail with only 1 fragment, you need 2 fragments. Therefore u will have to use a different graph in each one. because my problem is for example I have a list of recepies in my Master and once I click in a recepie it's shown the information of that recepie in the right side ( detail ).

**Comment by Zhuinden** [comment]
> Therefore u will have to use a different graph in each one. 

Not necessarily... I was putting off an example like this because I don't really use navigation, although I'm ok with it with the advances introduced in 2.2.0.

Maybe I'll consider writing that custom navigator implementation that would handle master-detail.

**Comment by MarcusWolschon** [comment]
> hey there!! I have one question, how do I pass data from the Master to the Detail fragment. 

A navigation graph as a whole can have arguments (that match arguments of the start-location).
